packages/google_maps_flutter/google_maps_flutter_ios/example/ios14/ios/RunnerTests/ExtractIconFromDataTests.m

```diff
@@ -0,0 +1,175 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+@import google_maps_flutter_ios;
+@import google_maps_flutter_ios.Test;
+@import XCTest;
+#import <OCMock/OCMock.h>
+
+@interface ExtractIconFromDataTests : XCTestCase
+- (UIImage *)createOnePixelImage;
+- (NSData *)convertImageToPNGData:(UIImage *)image;
+@end
+
+@implementation ExtractIconFromDataTests
+
+- (void)testExtractIconFromDataAssetAuto {
+  FLTGoogleMapMarkerController *instance = [[FLTGoogleMapMarkerController alloc] init];
+  NSObject<FlutterPluginRegistrar> *mockRegistrar =
+      OCMStrictProtocolMock(@protocol(FlutterPluginRegistrar));
+  id mockImageClass = OCMClassMock([UIImage class]);
+  UIImage *testImage = [self createOnePixelImage];
+
+  OCMStub([mockRegistrar lookupKeyForAsset:@"fakeImageNameKey"]).andReturn(@"fakeAssetKey");
+  OCMStub(ClassMethod([mockImageClass imageNamed:@"fakeAssetKey"])).andReturn(testImage);
+
+  NSArray *iconData = @[ @"asset", @"fakeImageNameKey", @"auto", @2 ];
+
+  UIImage *resultImage = [instance extractIconFromData:iconData registrar:mockRegistrar];
+  XCTAssertNotNil(resultImage);
+  [mockImageClass stopMocking];
```

Comment:
im curious - when `mockImageClass` is deallocated, does it auto call the `stopMocking`?

Reply:
The mock should automatically calls stopMocking during its own deallocation ([ref](https://ocmock.org/reference/#:~:text=The%20mock%20automatically%20calls%20stopMocking%20during%20its%20own%20deallocation.)).
I removed these `[mockImageClass stopMocking]` lines.